tempto-core/src/main/java/com/teradata/tempto/s3/S3Client.java

```diff
@@ -0,0 +1,384 @@
+     */
+    public void deleteFile(String path)
+    {
+        boolean isFolder = path.substring(path.length() - 1).equals("/");
+
+        // If not a folder, delete file; otherwise throw exception
+        if (!isFolder) {
+            deleteObject(path);
+        } else {
+            throw new RuntimeException("Folder was passed in to deleteFile: " + path);
+        }
+
+        logger.debug("Deleted file {} from S3", path);
+    }
+
+    /**
+     * PRIVATE METHODS
+     */
+
+    /**
+     * Method to get all object names in given path
+     *
+     * @param path  path to search
+     * @return      list of object names
+     */
+    public ArrayList<String> getObjects(String path)
+    {
+        ArrayList<String> objects = new ArrayList<String>();
+
+        // If no forward slash at end of path, add one
+        if (!path.substring(path.length() - 1).equals("/")) {
+            path += "/";
+        }
+
+        // Get all directory listings from client
+        ListObjectsV2Result listing = client.listObjectsV2(new ListObjectsV2Request()
+                .withBucketName(bucketName)
+                .withPrefix(path)
+                .withDelimiter("/"));
+
+        // Loop through directory listing and only add if no forward slash is present on last character
+        // (which indicates the directory itself)
+        for (S3ObjectSummary objectSummary : listing.getObjectSummaries()) {
+            if (!objectSummary.getKey().substring(objectSummary.getKey().length() - 1).equals("/")) {
+                objects.add(objectSummary.getKey());
+            }
+        }
+
+        return objects;
+    }
+
+    /**
+     * Method to get all object names in given path (recursively)
+     *
+     * @param path  path to search
+     * @return      list of object names
+     */
+    public ArrayList<String> getAllObjects(String path)
+    {
+        ArrayList<String> objects = new ArrayList<String>();
+
+        // If no forward slash at end of path, add one
+        if (!path.substring(path.length() - 1).equals("/")) {
+            path += "/";
+        }
+
+        // Get all directory listings from client
+        ListObjectsV2Result listing = client.listObjectsV2(new ListObjectsV2Request()
+                .withBucketName(bucketName)
+                .withPrefix(path));
+
+        // Loop through directory listing and only add if no forward slash is present on last character
+        // (which indicates the directory itself)
+        for (S3ObjectSummary objectSummary : listing.getObjectSummaries()) {
+            if (!objectSummary.getKey().equals(path)) {
+                objects.add(objectSummary.getKey());
+            }
+        }
+
+        return objects;
+    }
+
+    /**
+     * Method to deletes object
+     *
+     * @param path  path of object to delete
+     * @throws RuntimeException
+     */
+    private void deleteObject(String path)
+    {
+        boolean folder = path.substring(path.length() - 1).equals("/");
```

Comment:
determining if path points to folder or file just to have different exception is not worth an effort. Drop that.
And make exception message `Object does not exist`.

The other question is do we actually want to fail the `deleteObject` method if we remove inexistent object. This can make use of such method in cleanup procedures problematic.